packages/kbn-cell-actions/src/types.ts

```diff
@@ -92,18 +108,19 @@ export interface CellActionExecutionContext extends ActionExecutionContext {
   /**
    * Extra configurations for actions.
    */
-  metadata?: Record<string, unknown>;
+  metadata?: Metadata;
 }
 
-export interface CellActionCompatibilityContext extends ActionExecutionContext {
+export interface CellActionCompatibilityContext<M extends Metadata = Metadata>
+  extends ActionExecutionContext {
   /**
    * The object containing the field name and type, needed for the compatibility check
    */
-  field: Pick<CellActionField, 'name' | 'type'>;
+  field: Pick<CellActionField, 'name' | 'type' | 'aggregatable'>;
```

Comment:
I'd recommend adding `esTypes` - to check if it's a `keyword` in ES and isNestedField - to check if it's a nested type in ES. Lens only support `keyword` and `not nested` field, adding these two checks could avoid error when viewing in `Top N chart`. Relevant issue: https://github.com/elastic/kibana/issues/150158

Reply:
I created an issue for it. https://github.com/elastic/kibana/issues/150347

 Unfortunately, CellActions currently doesn't support `field?.subType`. We could temporarily add `subType` to the metadata. Let's discuss it in the new ticket.